hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/server/DefaultReadinessChecker.java

```diff
@@ -12,25 +19,40 @@
 import java.util.concurrent.ThreadFactory;
 import java.util.concurrent.TimeUnit;
 
-public class DefaultReadinessChecker implements ReadinessChecker {
+public class DefaultReadinessChecker implements ReadinessChecker, NodeCacheListener {
     private static final Logger logger = LoggerFactory.getLogger(DefaultReadinessChecker.class);
 
     private final boolean enabled;
+    private final boolean kafkaCheckEnabled;
     private final Duration interval;
     private final TopicMetadataLoadingRunner topicMetadataLoadingRunner;
-    private final ReadinessRepository readinessRepository;
     private final ScheduledExecutorService scheduler;
+    private final ObjectMapper mapper;
+    private final NodeCache cache;
 
+    private volatile boolean adminReady = false;
     private volatile boolean ready = false;
 
     public DefaultReadinessChecker(TopicMetadataLoadingRunner topicMetadataLoadingRunner,
-                                   ReadinessRepository readinessRepository,
+                                   CuratorFramework curator,
+                                   ZookeeperPaths paths,
+                                   ObjectMapper mapper,
                                    boolean enabled,
+                                   boolean kafkaCheckEnabled,
                                    Duration interval) {
         this.enabled = enabled;
+        this.kafkaCheckEnabled = kafkaCheckEnabled;
         this.interval = interval;
         this.topicMetadataLoadingRunner = topicMetadataLoadingRunner;
-        this.readinessRepository = readinessRepository;
+        this.mapper = mapper;
+        this.cache = new NodeCache(curator, paths.frontendReadinessPath());
+        cache.getListenable().addListener(this);
+        try {
+            cache.start(true);
+        } catch (Exception e) {
+            throw new InternalProcessingException("Readiness cache cannot start.", e);
+        }
+        refreshAdminReady();
```

Comment:
maybe move this method call to the `start` method ? Best practises say it is always better to call a method on the fully constructed object :)